### PR TITLE
fix: Connect smart speaker to open wifi

### LINKED ISSUE #1743: To allow the WiFi config to not have a password for networks without a password
**Actual Behaviour**
Right now, if we have an Open wifi network, we are unable to configure it with the Smart Speaker.

**Expected Behaviour**

To have a feature to allow connection to an Open Wifi Network

**Steps to reproduce it**

Try to connect the Speaker to an open network

**Would you like to work on the issue?**
No

### LINKED ISSUE #2268: Communication to ipinfo.io is not permitted by network policy
**Describe the bug**
Currently, there exist some network connection errors, while connecting to ipinfo.io. These errors are shown in the log.

**Expected Behaviour**
Such errors should not exist.

**Steps to reproduce it**
Open logcat at the android studio, now open the app.

**LogCat and Screenshot for the issue**
Screenshot 2019-06-17 at 11 29 05 AM

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | Redmi Note 5 Pro |
|Android Version| Pi 9.0  |

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #1743 
Fixes #2268

Changes:
Fixed the error that prevented the app to connect the smart speaker to open wifi.
Now users can connect to open wifi simply by clicking next in the wifi password prompt.

Screenshots for the change: 
ezgif com-video-to-gif

### CODE REVIEW

**Comment by atm1504** [comment]
Apk: https://drive.google.com/file/d/1-40PxYAb0gbM6o_VOJJpXdPHh6_BqKxL/view?usp=drivesdk

**Comment by norbusan** [comment]
I asked for access via my gmail account to the apk, thanks

**Comment by atm1504** [comment]
> I asked for access via my gmail account to the apk, thanks

I have updated the link. Please check now. It won't ask for permission. Sorry for the inconvenience caused.

**Comment by atm1504** [comment]
Apk: 
[app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3289517/app-fdroid-debug.apk.zip)

**Comment by norbusan** [comment]
@atm1504 same with the .apk from the zip. I unzipped it, downloaded it via dropbox to my phone, and tried to install it using x-plore (fs manager, I use often for isntalling apks).

This is Android 9, Huawei Mate 10 Pro.

**Comment by iamareebjamal** [comment]
@norbusan What failed while installing?

**Comment by norbusan** [comment]
@iamareebjamal How can I know? It starts installing, and then sqitches to a screen "Installation has failed" ...

**Comment by iamareebjamal** [comment]
@atm1504 Have you given unsigned APK?

**Comment by atm1504** [comment]
> @atm1504 Have you given unsigned APK?

No

**Comment by norbusan** [comment]
Hmmm, strange. Any tip how one can debug that?

**Comment by atm1504** [comment]
Can you please try out a different package installer other than the one you are using now. : )

**Comment by norbusan** [comment]
Re-downloaded the zip, tried with Huawei's "Files", and then again with Firefox downloader ... all with the same effect.

**Comment by iamareebjamal** [comment]
@atm1504 Come on, there is a single package manager on Android, there is no other option. You've got to know that

**Comment by atm1504** [comment]
> @atm1504 Come on, there is a single package manager on Android, there is no other option. You've got to know that
> 
> Did you download and try installing yourself?

Yes, I did so. It worked fine in both of the devices that I used.

**Comment by iamareebjamal** [comment]
The change in this PR has no relation to the issue it is solving

**Comment by atm1504** [comment]
> The change in this PR has no relation to the issue it is solving

I debugged the code. The error was mentioning about failing to connect because of clear text communication. Google search reveals that it occurs because of local connections. And we cannot use https in local, we always have http. To overcome this error, we need to take a special permission in manifest. That is what I added. Also, I faced similar issue in an app of my college and solved that in the same way. The app works on intranet of campus (http connection?)

**Comment by iamareebjamal** [comment]
The error you mentioned only happens on P+ Android devices. Is it working on lower versions.

Secondly, the solution is to whitelist IPs and not allow cleartext communication by default

**Comment by atm1504** [comment]
> The error you mentioned only happens on P+ Android devices. Is it working on lower versions.
> 
> Secondly, the solution is to whitelist IPs and not allow cleartext communication by default

I did found the error in oreo also.

Okk, so u mean what i should do is that only allow those particular ip's to have clear text communication, rather then allowing all.

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 @arundhati24 please review it.

File: app/src/main/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
+        <domain includeSubdomains="true">10.0.0.1:5000</domain>
+        <domain includeSubdomains="true">ipinfo.io</domain>
+        <domain includeSubdomains="true">api.duckduckgo.com</domain>
```
Comment (iamareebjamal) [inline app/src/main/res/xml/network_security_config.xml]: Remove last 2, replace with https

**Comment by iamareebjamal** [comment]
Have you now verified that it is working?

**Comment by atm1504** [comment]
Apk: [app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3306014/app-fdroid-debug.apk.zip)

@iamareebjamal I have tested this apk, after the change, it's working now.

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 @arundhati24  please review it. I cannot move further with pr #2267 until this is merged. : )

**Pull request merged**